docs/earthfile/earthfile.md

```diff
@@ -542,6 +542,25 @@ FROM +docker-image --NAME=john
 
 For more information on how to use build args see the [build arguments and secrets guide](../guides/build-args.md). A number of builtin args are available and are pre-filled by Earthly. For more information see [builtin args](./builtin-args.md).
 
+{% hint style='info' %}
+##### Shadowing Variables
+
+By default, `ARG` scoping isn't intuitive. When an `ARG` statement is parsed, earthly will look up _any_ previous declaration of the `ARG` and use the previous value, ignoring any new default. So for example:
+
+```
+VERSION 0.7
+ARG --global foo = bar
+
+baz:
+    ARG foo = bacon
+    RUN echo $foo
+```
+
+would print `bar`.
+
+Changing the `VERSION` line to `VERSION --arg-scope-and-set 0.7` changes this behavior, allowing local `ARG`s to shadow global `ARG`s and erroring if an `ARG` is redeclared in the same scope. This means that the above example would instead print `bacon`.
```

Comment:
This should state more clearly that this enables an experimental feature.

Reply:
Updated to mention that the flag is experimental. Thanks!